ref(grouping): Refactor similarity utils

armenzg opened the pull request:
It makes reading the code a little easier.

armenzg committed:
ref(grouping): Reduce code in function

Refactored three functions out of `get_stacktrace_string`.

armenzg commented on src/sentry/seer/similarity/utils.py:
```diff
@@ -197,59 +205,15 @@ def get_stacktrace_string(data: dict[str, Any], platform: str | None = None) ->
-
     metrics.distribution("seer.grouping.exceptions.length", len(exceptions))
 
-    def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
```
Now that we use a dictionary to track the metrics it is not as ugly to take this function out of here.

armenzg commented on src/sentry/seer/similarity/utils.py:
```diff
@@ -266,31 +230,52 @@ def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
 
         result_parts.append((header, frame_strings))
 
+    return generate_stacktrace_string(
```
All of this code has been moved within the function:
<img width="666" alt="image" src="https://github.com/user-attachments/assets/061d868f-d891-42ed-9df1-cf85fd3abbfe">

armenzg committed:
Minor touches

armenzg committed:
Merge branch 'master' into ref/similarity_utils/armenzg-2

armenzg committed:
Fix bug

armenzg committed:
Fix test

jangjodi reviewed (approved):
this looks great! just letting you know that I added another frame metric [here](https://github.com/getsentry/sentry/pull/81700/files) that will need to be added to the `FrameMetrics` dictionary

armenzg committed:
Merge branch 'master' into ref/similarity_utils/armenzg-2

armenzg committed:
Minor

armenzg commented:
> this looks great! just letting you know that I added another frame metric [here](https://github.com/getsentry/sentry/pull/81700/files) that will need to be added to the `FrameMetrics` dictionary

Thanks for letting me know! I added it.
